### DISCUSSION TITLE
speed up bacpac import for small files

### DISCUSSION THREAD

**Discussion by oysbre** [d365collaborative/d365fo.tools#759] [Ideas]
by adding /p:DisableIndexesForDataPhase=FALSE as parameter for sqlpackage, the data is loaded without enable/disabling indexes first. when using small bacpac file the time is almost the half. https://denistrunin.com/performance-restoretier2/
is there a way to either check the size of the bacpac file or just add the option for the parameter in invoke-sqlpackage ?

**Comment by FH-Inway** [comment]
Thanks for the suggestion. @TrudAX also suggested this and other parameters in #648 . We did not get around to implementing any of it yet, but maybe this gets things going again.

Would you be interested in working with us to implement this?

**Reply by TrudAX** [reply]
I don't know PowerShell(can only ask chatGPT), but I can be involved in testing/discussion. For this one - I suggest you just add a parameter, project DB size is quite stable, so people can just decide what to use(size check will be overkill IMHO)

**Comment by FH-Inway** [comment]
@oysbre Take a look at https://github.com/d365collaborative/d365fo.tools/issues/648#issuecomment-1752087512
